lib/fol/Variable.h

```diff
@@ -3,5 +3,13 @@
 #include "Symbol.h"
 
 namespace fol {
-    struct Variable : Symbol {};
+    struct Variable : Symbol {
+        friend bool operator==(const Variable &lhs, const Variable &rhs) {
+                bool eq=false;
```

Comment:
Can lines 9 - 12 be shrunk to this single line?

```
return lhs.name == rhs.name
```

Reply:
Yes, true!